http/get_simple/java/server/src/main/java/com/example/ArrowHttpServer.java

```diff
@@ -0,0 +1,131 @@
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.memory.ArrowBuf;
+import org.apache.arrow.vector.BigIntVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.VectorLoader;
+import org.apache.arrow.vector.VectorUnloader;
+import org.apache.arrow.vector.ipc.ArrowStreamWriter;
+import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Schema;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class ArrowHttpServer extends AbstractHandler {
+
+    static BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
```

Comment:
Any special reason why this is a static member? Also should we use with try-with-resources alternatively to release the resources gracefully?

Reply:
It seems like the proper way should be to override destroy and handle this as an instance member

Reply:
I made everything static for convenience and simplicity (since non-static members can't be referenced in a static context). I'm happy to change this but it's unclear to me why and how.

I wanted to reuse the same allocator everywhere here (because I think it doesn't work if you use different allocators) but I don't know how to pass it into `handle` except by making it a member of the class, and when it's a member of the class I don't think there's any way to use try-with-resources.

Reply:
This is OK to have but I think we'd want to figure this out somehow.

You wouldn't use try-with-resources. You would use whatever initialize/cleanup methods that Jetty provides (it appears AbstractHandler implements something called Destroyable).

Reply:
I added a note in the README that says
> For simplicity, the example here uses static class members and does not properly initialize and release resources or handle errors.